src/supplemental/nanolib/conf_ver2.c

```diff
@@ -381,6 +381,21 @@ conf_basic_parse_ver2(conf *config, cJSON *jso)
 			}
 		}
 	}
+	cJSON *jso_filetransfer = cJSON_GetObjectItem(jso, "filetransfer");
+	if (jso_filetransfer) {
+		conf_filetransfer *filetransfer = &(config->filetransfer);
+
+		hocon_read_bool(filetransfer, enable, jso_filetransfer);
+		char *url = cJSON_GetStringValue(hocon_get_obj("bind", jso_filetransfer));
+		if (url != NULL) {
+			hocon_read_address_base(
+				filetransfer, url, "bind", "mqtt-tcp://", jso_filetransfer);
+		}
+		char *topic = cJSON_GetStringValue(hocon_get_obj("topic", jso_filetransfer));
```

Comment:
No need to check if `topic` is NULL here. The `hocon_read_str` macro will handle the check, and if `topic` is NULL, it will automatically use the default value.

Reply:
compiler will take care of this